### ISSUE TITLE
False positive on yoda comparisons

### ISSUE THREAD

**Issue by JurianArie** [slevomat/coding-standard#1333]
Version: 7.0.18
PHP version: 8.0.15

I am getting false reports of breaking the yoda comparison rule.

```php
<?php

function test($condition, $actual, $expected)
{
    return match ($condition) {
        '=' => $actual == $expected,
        '>' => $actual > $expected,
        '>=' => $actual >= $expected,
        '<' => $actual < $expected,
        '<=' => $actual <= $expected,
        '!=' => $actual != $expected,
        default => false,
    };
}

```

The following error is reported: `11 | ERROR | [x] Yoda comparisons are disallowed.`

When running phpcbf it breaks the code:

```php
<?php

function test($condition, $actual, $expected)
{
    return match ($condition) {
        '=' => $actual == $expected,
        '>' => $actual > $expected,
        '>=' => $actual >= $expected,
        '<' => $actual < $expected,
        '<=' => $actual <= $expected,
        $expected != '!=' => $actual,
        default => false,
    };
}

```

**Comment by kukulich** [comment]
Thank you for your report.

Fixed in https://github.com/slevomat/coding-standard/commit/c611be7a29ffc254e967b3ed96ea0c9e4d162c35
